airflow/configuration.py

```diff
@@ -1260,18 +1260,28 @@ def get_custom_secret_backend() -> Optional[BaseSecretsBackend]:
 def initialize_secrets_backends() -> List[BaseSecretsBackend]:
     """
     * import secrets backend classes
+    * sort by prioritisation
     * instantiate them and return them in a list
     """
     backend_list = []
 
+    for class_name in DEFAULT_SECRETS_SEARCH_PATH:
+        secrets_backend_cls = import_string(class_name)
+        backend_list.append(secrets_backend_cls())
+
     custom_secret_backend = get_custom_secret_backend()
 
     if custom_secret_backend is not None:
-        backend_list.append(custom_secret_backend)
+        backend_priority = conf.get('secrets', 'backend_priority', fallback='') or 'HIGH'
 
-    for class_name in DEFAULT_SECRETS_SEARCH_PATH:
-        secrets_backend_cls = import_string(class_name)
-        backend_list.append(secrets_backend_cls())
+        # Do enum validation here, because other enums checks run after initialise custom secrets backends
+        if backend_priority not in CustomSecretsBackendPriority.priorities():
+            raise AirflowConfigException(
+                f"`['secrets'] backend_priority` should not be {backend_priority!r}. "
+                f"Possible values: {', '.join(CustomSecretsBackendPriority.priorities())}."
+            )
+
+        backend_list.insert(CustomSecretsBackendPriority[backend_priority], custom_secret_backend)
```

Comment:
I wonder if we should have a `conf.getenum` for this pattern. If not, an `upper` call here would save some unnecessary user frustration, at least.

Reply:
added `upper`